bot/exts/moderation/defcon.py

```
@@ -28,71 +36,75 @@
+    )
 
 
 class Defcon(Cog):
     """Time-sensitive server defense mechanisms."""
 
-    days = None  # type: timedelta
-    enabled = False  # type: bool
+    redis_cache = RedisCache()
 
     def __init__(self, bot: Bot):
         self.bot = bot
         self.channel = None
-        self.days = timedelta(days=0)
+        self.threshold = relativedelta(days=0)
+        self.expiry = None
+
+        self.scheduler = Scheduler(self.__class__.__name__)
 
-        self.bot.loop.create_task(self.sync_settings())
+        self.bot.loop.create_task(self._sync_settings())
 
     @property
     def mod_log(self) -> ModLog:
         """Get currently loaded ModLog cog instance."""
         return self.bot.get_cog("ModLog")
 
-    async def sync_settings(self) -> None:
+    @redis_cache.atomic_transaction
+    async def _sync_settings(self) -> None:
         """On cog load, try to synchronize DEFCON settings to the API."""
+        log.trace("Waiting for the guild to become available before syncing.")
         await self.bot.wait_until_guild_available()
         self.channel = await self.bot.fetch_channel(Channels.defcon)
 
-        try:
-            response = await self.bot.api_client.get('bot/bot-settings/defcon')
-            data = response['data']
+        log.trace("Syncing settings.")
 
-        except Exception:  # Yikes!
+        try:
+            settings = await self.redis_cache.to_dict()
+            self.threshold = parse_duration_string(settings["threshold"])
+            self.expiry = datetime.fromisoformat(settings["expiry"]) if settings["expiry"] else None
+        except Exception:
             log.exception("Unable to get DEFCON settings!")
-            await self.bot.get_channel(Channels.dev_log).send(
-                f"<@&{Roles.admins}> **WARNING**: Unable to get DEFCON settings!"
-            )
+            await self.channel.send(f"<@&{Roles.moderators}> **WARNING**: Unable to get DEFCON settings!")
```

I feel like this should also ping the DevOps since they would be more capable of knowing why we can't connect to Redis, and two of our DevOps members aren't mods.

---

Agreed

---

Also looking back at it, I think the message should contain the traceback.

---

Doesn't seem right for the defcon channel. Maybe devops? but then again it might be better to just re-raise the exception

---

I wouldn't mind having it in `#defcon` to be honest. The issue with re-raising is that it might get lost in Sentry because the issue already exists, although the traceback will show in the logs.